src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -79,4 +80,11 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
     protected abstract String parserName();
 
     protected abstract void doXContent(XContentBuilder builder, Params params) throws IOException;
+
+    @Override
+    public ActionRequestValidationException validate() {
+        // default impl does not validate, subclasses should override.
+        // TODO Remove once all queries support validation
```

Comment:
can you use //norelease instead of TODO? then we are sure we don't release if this method is there cause the build script stops the build in that case